TicTacToe/GamePlay.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+
+namespace TicTacToe
+{
+    public class GamePlay
+    {
+        private int _player;
+        private GameStatus _gameStatus;
+        private readonly Board _board;
+        private readonly int[,] _boardArray;
+        private bool _givenUp;
+        
+        public GamePlay()
+        {
+            _board = new Board();
+            _boardArray = _board.GetBoard();
+            _player = RandomlyPickPlayerForFirst();
+            _givenUp = false;
+            _gameStatus = GameStatus.Playing;
+        }
+
+        public void Play()
+        {
+            var uiConsole = new UserInputConsole();
+            while (_gameStatus is GameStatus.Playing)
+            {
+                uiConsole.OutputBoard(_boardArray);
+                uiConsole.UpdatePlayerInput(_player);
+                if (uiConsole.PlayerHasGivenUp())
+                {
+                    _givenUp = true;
+                }
+                else
+                {
+                    var inputMove = uiConsole.GetPlayerMove();
+                    string outputMessage = HandleMoveInput(inputMove);
+                    uiConsole.OutputMessage(outputMessage);
+                }
+                UpdateGameStatus();
+            }
+            var finalMessage = HandleEndOfGame();
+            uiConsole.OutputMessage(finalMessage);
+            uiConsole.OutputBoard(_boardArray);
+        }
+        
+        public int GetPlayer()
+        {
+            return _player;
+        }
+        
+        public void SwapPlayer()
+        {
+            _player = _player == Constants.PlayerXValue ? Constants.PlayerOValue : Constants.PlayerXValue;
+        }
+        
+        public GameStatus GetCurrentStatus()
```

Comment:
I'm not sure GameStatus needs to be public, is there anything else that needs to know the GameStatus?